crates/biome_js_analyze/src/lint/correctness/no_unused_private_class_members.rs

```diff
@@ -160,6 +249,45 @@ fn traverse_members_usage(
     private_members.into_iter().collect()
 }
 
+/// Check if a TsPropertyParameter is also unused as a function parameter
+fn check_ts_property_parameter_usage(
+    ctx: &RuleContext<NoUnusedPrivateClassMembers>,
+    ts_property_param: &TsPropertyParameter,
+) -> bool {
+    if let Ok(AnyJsFormalParameter::JsFormalParameter(param)) = ts_property_param.formal_parameter()
+    {
+        if let Ok(binding) = param.binding() {
+            if let Some(identifier_binding) = binding
+                .as_any_js_binding()
+                .and_then(|b| b.as_js_identifier_binding())
+            {
```

Comment:
This would benefit from `let` chaining:

```suggestion
    if let Ok(AnyJsFormalParameter::JsFormalParameter(param)) = ts_property_param.formal_parameter()
        && let Ok(binding) = param.binding()
        && let Some(identifier_binding) = binding
            .as_any_js_binding()
            .and_then(|b| b.as_js_identifier_binding())
    {
```